src/services/media-library-service.js

```diff
@@ -116,6 +108,29 @@ class MediaLibraryService {
     const result = await this.mediaLibraryItemStore.getMediaLibraryItemTagsMatchingText(sanitizedSearchString);
     return result[0]?.uniqueTags || [];
   }
+
+  async _getSearchableMediaLibraryItems({ textQuery, resourceTypes }) {
+    if (!textQuery.isValid) {
+      return [];
+    }
+
+    const textAndResourceTypeQueryConditions = [textQuery.query];
+    if (resourceTypes) {
+      textAndResourceTypeQueryConditions.push({ resourceType: { $in: resourceTypes } });
+    }
+
+    const mediaLibraryItems = await this.mediaLibraryItemStore.getMediaLibraryItemsByConditions(textAndResourceTypeQueryConditions);
+
+    const positiveTokensArray = [...textQuery.positiveTokens].filter(token => token.toLowerCase());
+    return mediaLibraryItems
+      .map(item => {
+        const exactTagMatchCount = item.tags.filter(tag => textQuery.positiveTokens.has(tag.toLowerCase())).length;
+        const partialNameMatchCount = positiveTokensArray.filter(token => item.name.toLowerCase().includes(token)).length;
```

Comment:
Now we also calculate a relevance based on the file name even for the case when we only search by tags, as this function is used in both `getSearchableMediaLibraryItemsByTags` and `getSearchableMediaLibraryItemsByTagsOrName`. Maybe we could refactor this function to be like `_getSearchableMediaLibraryItems({ textQuery, resourceTypes, searchAlsoInNames = false })` and, depending on that, the function does the query and relevance calculation?

Reply:
Makes sense, corrected in this way, thanks!

Reply:
Nice, looks good!